library/general/src/lib/ui/installation/layout.rb

```diff
@@ -0,0 +1,166 @@
+# Copyright (c) [2020] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "yast"
+require "ui/installation/layout_config"
+
+module UI
+  module Installation
+    # Class to define the wizard layout for the installation
```

Comment:
is it really only for installation? Or it can be used also for firstboot or even for common wizards?

Reply:
Well, it could be used at every place where a wizard can be used, but it is mainly designed for installation mode (basically installer and firstboot). I got inspired by *UI::InstallationDialog*. Some layout elements like the top banner or the sidebar with steps or placing the title on the left are only used by YasT in the installer and firstboot clients. For that reason this class is focused on configuring an "installation" layout. Moreover, the provided stylesheets usually only take care of such elements during the installation/firstboot (installation.qss). IMHO, opening it to a more generic layout might make confusion.

Reply:
I am just a bit worried if we by naming not limit its future usage when there is no reason. What about namespace `Wizards`? So it will be `UI::Wizards::Layout` ? So it also clearly indicate that it is for wizard like dialogs.

Reply:
Yes, thinking twice about, you are right. Although the idea was born for installation/firstboot, I don't see any reason to limit it. And yes, `UI::Wizards::Layout` looks a good place. Thanks!